lib/id3c/cli/redcap.py

```diff
@@ -225,7 +238,43 @@ def records(self, *,
         if filter is not None:
             parameters['filterLogic'] = str(filter)
 
-        return [Record(self, r) for r in self._fetch("record", parameters)]
+        if page_size is not None:
+            return self._fetch_records_paged(parameters, page_size)
+        else:
+            return list(self._fetch_records(parameters))
+
+
+    def _fetch_records_paged(self, parameters: dict, page_size: int) -> Iterator['Record']:
+        assert bool(self._details["record_autonumbering_enabled"]), \
+            "Record auto-numbering must be enabled to use page_size parameter"
+
+        next_record_id = self._fetch("generateNextRecordName")
```

Comment:
I didn't know where the "generateNextRecordName" came from, so I did some Googling. Ah, this is a string that the REDCap API understands. Could you put a comment above this line with something like the following

"Generate Next Record Name (content=generateNextRecordName) - To be used by projects with record auto-numbering enabled, this method exports the next potential record ID for a project. It generates the next record name by determining the current maximum numerical record ID and then incrementing it by one."

I see this on https://research.fredhutch.org/cds/en/redcap-version-history.html

Reply:
IIRC, the tough part about linking to our official redcap docs is that the link is version dependent and break once theres a new version released e.g. https://redcap.iths.org/redcap_v10.5.2/API/help.php?content=exp_next_id will soon break

Reply:
Sure. I was suggesting putting in a comment to say what generateNextRecordName does.

Reply:
I will add a comment.

I didn't think one was necessary since `self._fetch(…)` is used elsewhere as well, without similar comments.

Reply:
Thanks